crates/nu-json/src/ser.rs

```diff
@@ -1034,6 +1034,23 @@ where
 {
     let vec = to_vec(value)?;
     let string = String::from_utf8(vec)?;
-    let output = string.lines().map(str::trim).collect();
+
+    let mut inside_quotes = false;
+    let mut last_char = 'p'; // 'p' is just a placeholder, can be anything except for '\\'
+    let output = string
+        .chars()
```

Comment:
I'm wondering if `chars()` could be an issue with unicode characters. Would `graphemes()` be better here?

Reply:
In my understanding it shouldn't be a problem, since the only character that is modified is the whitespace which is checked for with the `.is_whitespace()` function that also considers unicode characters.

Reply:
The only whitespace that should be removed is outside of quotes/strings, which means that it is whitespace added by the encoder. This should only be ever be spaces, tabs, etc. which is covered by `char::is_whitespace`.
So, I think `graphemes` is unnecessary.

Reply:
In fact, could you change it to [`is_ascii_whitespace`](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_whitespace) instead? The encoder only emits whitespace characters covered by this. Thanks!